contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,78 @@
+As we work on the maturity of our components, we will sometimes need to build components in a parallel track/bundle without breaking existing components. Eventually, the new components would replace the old ones in the main bundle and this document explains the process around it.
+
+## Developing a draft component
```

Comment:
```suggestion
## Developing a replacement component
```

We're planning to consolidate the "draft" and "experimental" lifecycle stages into one stage (https://github.com/github/primer/issues/1463), so I'm suggesting this change for clarity.